torch/utils/cpp_extension.py

```diff
@@ -546,29 +579,38 @@ def build_extensions(self) -> None:
         compiler_name, compiler_version = self._check_abi()
 
         cuda_ext = False
+        sycl_ext = False
         extension_iter = iter(self.extensions)
         extension = next(extension_iter, None)
-        while not cuda_ext and extension:
+        while (not cuda_ext and not sycl_ext) and extension:
             for source in extension.sources:
                 _, ext = os.path.splitext(source)
                 if ext == '.cu':
                     cuda_ext = True
+                elif ext == '.sycl':
+                    sycl_ext = True
+
+                if cuda_ext and sycl_ext:
```

Comment:
```suggestion
                if cuda_ext or sycl_ext:
```
?

Reply:
That was not a typo. The reason for  `and` was to support a potential case when someone builds an extension which mixes cuda and sycl sources. In such a case we are looking for _all_ known extensions in this loop and condition is to stop looking once there is no more need to continue the loop, i.e. we found that both are used.

If you believe mix of cuda/sycl sources is too artificial to support - I can change to `or`. I in any case will update and add a comment describing the reasoning behind the condition.

Reply:
Yeah I was assuming you never have both haha
This check is inconsistent with the while loop condition though. If you want to handle having both, the while condition needs updating. Otherwise this one can be relaxed.

Reply:
I fixed condition in `while` and added a comment. Hope that's correct now. Let me know if not :).